### PR TITLE
force workflow deletion.

### PR DESCRIPTION
**Pull request by manan164**
Have an environment level flag to decide on Redis archival.

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java [hunk 1]
```diff
@@ -661,7 +661,7 @@ void completeWorkflow(Workflow wf) {
         Monitors.recordWorkflowCompletion(workflow.getWorkflowName(), workflow.getEndTime() - workflow.getStartTime(), wf.getOwnerApp());
         queueDAO.remove(DECIDER_QUEUE, workflow.getWorkflowId());    //remove from the sweep queue
 
-        if (workflow.getWorkflowDefinition().isWorkflowStatusListenerEnabled()) {
+        if (config.getForceWorkflowDeletionFromRedis() || workflow.getWorkflowDefinition().isWorkflowStatusListenerEnabled()) {
```
Comment (mudit3774) [inline core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: Consider having a different approach in which you are not blocking this onWorkflowCompleted event if ForceWorkflowDeletionFromRedis is not setup. 

We can do various other functions onWorkflowComplete.

File: core/src/main/java/com/netflix/conductor/core/config/Configuration.java [hunk 2]
```diff
@@ -154,6 +156,9 @@ default boolean getBoolProperty(String name, boolean defaultValue) {
         return null;
     }
 
+    default boolean getForceWorkflowDeletionFromRedis() {
```
Comment (mudit3774) [inline core/src/main/java/com/netflix/conductor/core/config/Configuration.java]: Consider not having the DAO name in the function -- "fromRedis" might not be valid.

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java [hunk 3]
```diff
@@ -758,7 +758,7 @@ public void terminateWorkflow(Workflow workflow, String reason, String failureWo
         // Send to atlas
         Monitors.recordWorkflowTermination(workflow.getWorkflowName(), workflow.getStatus(), workflow.getOwnerApp());
 
-        if (workflow.getWorkflowDefinition().isWorkflowStatusListenerEnabled()) {
```
Comment (mudit3774) [inline core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: Consider removing this check also. We might want to do multiple actions on worklfow termination.

Comment (manan164) [reply core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: Why we should remove this? I mean do you mean that now only force.workflow.deletion will be a feature gate?

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java (hunk 1 above)
Comment (manan164) [reply core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: yeah.

**Pull request merged**